### DISCUSSION TITLE
Shorten silence URL

### DISCUSSION THREAD

**Discussion by a-mazed** [grafana/grafana#39043] [Alerting]
Please add an option to shorten silence URL.

Right now, my silence URL in Slack alert message can be 560 characters long, and that's not the biggest value. It's not good at all.

Grafana v8.1.3 using NGAlert

**Comment by haelekuin** [comment]
Sorry for reviving an old discussion topic here, but I stumbled upon this while looking to shorten the URL myself! I agree that the URL gets pretty long. For some of our notifications they'd run into message length limits because of them.

My workaround for now is to just turn it into a hyperlink. For Slack, this just needs something like this in the notification template

```go
{{ if gt (len .SilenceURL ) 0 }}
<{{ .SilenceURL | safeHtml }}|Silence Alert>{{ end }}
``` 
The output is significantly more readable:
![1687955739](https://github.com/grafana/grafana/assets/86392447/f4af2b67-b503-4293-994d-b69cba8397e8)

**Reply by tommyreins** [reply]
which field of the alert rule did you put that code?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
